plugins/identity-shiro/src/main/java/org/opensearch/identity/shiro/ShiroTokenManager.java

```diff
@@ -75,6 +75,19 @@ public AuthToken issueOnBehalfOfToken(Subject subject, OnBehalfOfClaims claims)
         return token;
     }
 
+    @Override
+    public AuthToken issueServiceAccountToken(String audience) {
+
+        String password = generatePassword();
+        final byte[] rawEncoded = Base64.getEncoder().encode((audience + ":" + password).getBytes(UTF_8)); // Make a new
```

Comment:
Elsewhere in the code (I think [UUID generation](https://github.com/opensearch-project/OpenSearch/blob/main/libs/common/src/main/java/org/opensearch/common/RandomBasedUUIDGenerator.java#L60C19-L60C19)) we've used URL-safe encoders.  Since this will be passed in headers I think that's needed here (and possibly when decoding?)

Reply:
Oh okay, I will make the swap.